src/exponential_backoff.js

```diff
@@ -0,0 +1,43 @@
+import _ from 'underscore';
+import log from './log';
+import { delayPromise } from './utilities';
+
+export class RetryableError extends Error {}
```

Comment:
Actually, I just figured one thing - the `RetryableError` instance should have a constructor with a single parameter - the underlying `Error` object. That's what we should rethrow. Many underlying errors cannot change their type to `RetryableError` so we'd loose the stack trace etc. if we just copied the message.